### PR TITLE
GEOPY-90

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-90 - Support of geoh5 MT object in geoh5py**

### CODE REVIEW

File: geoh5py/objects/surveys/magnetotellurics.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        :return: List of new Data objects.
+        """
+        data_objects = []
+        if self.channels is None or not self.channels:
+            raise AttributeError(
+                "The 'channels' property defining frequencies must be set before adding data."
+            )
+
+        if not isinstance(data, dict):
+            raise TypeError(
```
Comment (benk-mira) [inline geoh5py/objects/surveys/magnetotellurics.py:179]: Would be nice to put a more thorough validator here to check the component names, nesting structure, etc.  Maybe once we get the validator module moved over?

Comment (benk-mira) [reply geoh5py/objects/surveys/magnetotellurics.py:179]: Oh, I see, you're doing the extra level validation in the for loop..  Great!

File: tests/create_survey_mt_test.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        mt_survey = Magnetotellurics.create(workspace, vertices=vertices, name=name)
+
+        for key, value in {
+            "input_type": "Rx Only",
+            "survey_type": "Magnetotellurics",
+            "unit": "Hertz (Hz)",
+        }.items():
+            assert getattr(mt_survey, key) == value, f"Error setting defaults for {key}"
+
+        with pytest.raises(TypeError):
```
Comment (benk-mira) [inline tests/create_survey_mt_test.py:60]: Nice addition!  These can be a little stronger if you check that the output is what you programmed it to be.  Rules out a TypeError popping up in an unexpected place before your code hits:

with pytest.raises(TypeError) as excinfo:
    call_method()

assert "some string from my custom exception" in str(excinfo)

**Review by benk-mira — DISMISSED** [review]
Looks great!  Nothing blocking in my comments.  Address as you see fit.

File: tests/create_survey_mt_test.py (hunk 2 above)
Comment (domfournier) [reply tests/create_survey_mt_test.py:60]: Done

**Review by benk-mira — DISMISSED** [review]
Yep yep!

File: tests/create_survey_mt_test.py (hunk 2 above)
Comment (benk-mira) [reply tests/create_survey_mt_test.py]: Sweeeet!

**Pull request merged**